src/main/java/com/icthh/xm/ms/entity/service/metrics/MetricsAdapterImpl.java

```diff
@@ -0,0 +1,73 @@
+package com.icthh.xm.ms.entity.service.metrics;
+
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Slf4j
+@Service
+public class MetricsAdapterImpl implements MetricsAdapter {
+    private static final String DEFAULT_NAME = "defaultName";
+    private static final String DEFAULT_DESC = "Default counter";
+    private final MeterRegistry meterRegistry;
+
+    @Autowired
+    public MetricsAdapterImpl(MeterRegistry meterRegistry) {
+        this.meterRegistry = meterRegistry;
+    }
+
+    public void inc(String[] tags, Double amount, String name, String description) {
+        log.debug("Inc : {}, {}, {}, {}", tags, amount, name, description);
+        try {
+            validation(tags, name, description, amount);
+            if (tags == null || tags.length == 0) {
+                throw new IllegalArgumentException("List of tags is empty");
+            }
```

Comment:
why do not put this into validation method as well?

Reply:
done